Exception from passing test SslSelectChannelTimeoutTest.testMaxIdleNothingSent()

```
[AdvancedRunner] Running org.eclipse.jetty.server.ssl.SslSelectChannelTimeoutTest.testMaxIdleNothingSent()
Running org.eclipse.jetty.server.ssl.SslSelectChannelTimeoutTest.testMaxIdleNothingSent()
2018-05-16 14:46:41.022:INFO:oejs.Server:main: jetty-9.4.11-SNAPSHOT; built: 2018-05-16T04:43:12.635Z; git: 820b5752d7207fc0ed5294d13887780f7b28a48c; jvm 9.0.1+11
2018-05-16 14:46:41.026:INFO:oejus.SslContextFactory:main: x509=X509@6b4a2887(jetty,h=[jetty.eclipse.org],w=[]) for SslContextFactory@23b087c3[provider=null,keyStore=file:///usr/local/home/gregw/src/jetty-9.4.x/jetty-server/src/test/resources/keystore,trustStore=file:///usr/local/home/gregw/src/jetty-9.4.x/jetty-server/src/test/resources/keystore]
2018-05-16 14:46:41.027:INFO:oejus.SslContextFactory:main: x509=X509@237c8718(mykey,h=[],w=[]) for SslContextFactory@23b087c3[provider=null,keyStore=file:///usr/local/home/gregw/src/jetty-9.4.x/jetty-server/src/test/resources/keystore,trustStore=file:///usr/local/home/gregw/src/jetty-9.4.x/jetty-server/src/test/resources/keystore]
2018-05-16 14:46:41.030:INFO:oejs.AbstractConnector:main: Started ServerConnector@325e03b{SSL,[ssl, http/1.1]}{0.0.0.0:41871}
2018-05-16 14:46:41.032:INFO:oejs.Server:main: Started @35514ms
javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
	at java.base/sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1058)
	at java.base/sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	at java.base/sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1402)
	at java.base/sun.security.ssl.SSLSocketImpl.bytesInCompletePacket(SSLSocketImpl.java:907)
	at java.base/sun.security.ssl.AppInputStream.read(AppInputStream.java:144)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:185)
	at org.eclipse.jetty.util.IO.copy(IO.java:225)
	at org.eclipse.jetty.util.IO.copy(IO.java:135)
	at org.eclipse.jetty.util.IO.toString(IO.java:318)
	at org.eclipse.jetty.util.IO.toString(IO.java:290)
	at org.eclipse.jetty.server.ConnectorTimeoutTest.testMaxIdleNothingSent(ConnectorTimeoutTest.java:636)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
	at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:156)
	at java.base/sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1031)
	... 25 more
2018-05-16 14:46:43.439:INFO:oejs.AbstractConnector:main: Stopped ServerConnector@325e03b{SSL,[ssl, http/1.1]}{0.0.0.0:0}

```

I see the same stacktrace on OSX.

Note: this test does not fail, it seems that this failure isn't crucial to the testcase.

```
Running org.eclipse.jetty.server.ssl.SslSelectChannelTimeoutTest.testMaxIdleNothingSent()
2018-05-16 06:40:09.117:INFO:oejs.Server:main: jetty-9.4.11-SNAPSHOT; built: 2018-05-16T11:27:23.267Z; git: 4cc8f4dcf6c64eade4d34b2456061a9b243f002b; jvm 9.0.4+11
2018-05-16 06:40:09.118:INFO:oejus.SslContextFactory:main: x509=X509@2411cf8b(jetty,h=[jetty.eclipse.org],w=[]) for SslContextFactory@54c3bae[provider=null,keyStore=file:///Users/joakim/Code/Jetty/jetty.project/jetty-server/src/test/resources/keystore,trustStore=file:///Users/joakim/Code/Jetty/jetty.project/jetty-server/src/test/resources/keystore]
2018-05-16 06:40:09.118:INFO:oejus.SslContextFactory:main: x509=X509@6fad6743(mykey,h=[],w=[]) for SslContextFactory@54c3bae[provider=null,keyStore=file:///Users/joakim/Code/Jetty/jetty.project/jetty-server/src/test/resources/keystore,trustStore=file:///Users/joakim/Code/Jetty/jetty.project/jetty-server/src/test/resources/keystore]
2018-05-16 06:40:09.119:INFO:oejs.AbstractConnector:main: Started ServerConnector@68a0a4ae{SSL,[ssl, http/1.1]}{0.0.0.0:56584}
2018-05-16 06:40:09.119:INFO:oejs.Server:main: Started @229682ms
javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
	at java.base/sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1058)
	at java.base/sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	at java.base/sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1402)
	at java.base/sun.security.ssl.SSLSocketImpl.bytesInCompletePacket(SSLSocketImpl.java:907)
	at java.base/sun.security.ssl.AppInputStream.read(AppInputStream.java:144)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:185)
	at org.eclipse.jetty.util.IO.copy(IO.java:225)
	at org.eclipse.jetty.util.IO.copy(IO.java:135)
	at org.eclipse.jetty.util.IO.toString(IO.java:318)
	at org.eclipse.jetty.util.IO.toString(IO.java:290)
	at org.eclipse.jetty.server.ConnectorTimeoutTest.testMaxIdleNothingSent(ConnectorTimeoutTest.java:636)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
	at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:156)
	at java.base/sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1031)
	... 25 more
2018-05-16 06:40:11.527:INFO:oejs.AbstractConnector:main: Stopped ServerConnector@68a0a4ae{SSL,[ssl, http/1.1]}{0.0.0.0:0}
```

--- a/jetty-server/src/test/java/org/eclipse/jetty/server/ConnectorTimeoutTest.java
+++ b/jetty-server/src/test/java/org/eclipse/jetty/server/ConnectorTimeoutTest.java
@@ -43,6 +43,8 @@ import org.eclipse.jetty.server.handler.AbstractHandler;
 import org.eclipse.jetty.toolchain.test.AdvancedRunner;
 import org.eclipse.jetty.toolchain.test.TestTracker;
 import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
 import org.eclipse.jetty.util.log.StacklessLogging;
 import org.hamcrest.Matchers;
 import org.junit.Assert;
@@ -55,6 +57,8 @@ import org.junit.runner.RunWith;
 @RunWith(AdvancedRunner.class)
 public abstract class ConnectorTimeoutTest extends HttpServerTestFixture
 {
+    protected static final Logger LOG = Log.getLogger(ConnectorTimeoutTest.class);
+
     @Rule
     public TestTracker tracker = new TestTracker();
     
@@ -610,11 +614,17 @@ public abstract class ConnectorTimeoutTest extends HttpServerTestFixture
         }
         catch(SSLException e)
         {
-            e.printStackTrace();
+            if(LOG.isDebugEnabled())
+                LOG.debug(e);
+            else
+                LOG.info(e.getMessage());
         }
         catch(Exception e)
         {
-            e.printStackTrace();
+            if(LOG.isDebugEnabled())
+                LOG.debug(e);
+            else
+                LOG.info(e.getMessage());
         }
         Assert.assertTrue(TimeUnit.NANOSECONDS.toMillis(System.nanoTime()) - start < maximumTestRuntime);
 
@@ -639,11 +649,17 @@ public abstract class ConnectorTimeoutTest extends HttpServerTestFixture
         }
         catch(SSLException e)
         {
-            e.printStackTrace();
+            if(LOG.isDebugEnabled())
+                LOG.debug(e);
+            else
+                LOG.info(e.getMessage());
         }
         catch(Exception e)
         {
-            e.printStackTrace();
+            if(LOG.isDebugEnabled())
+                LOG.debug(e);
+            else
+                LOG.info(e.getMessage());
         }
         Assert.assertTrue(TimeUnit.NANOSECONDS.toMillis(System.nanoTime()) - start < maximumTestRuntime);
 

--- a/jetty-server/src/test/java/org/eclipse/jetty/server/ServerConnectorTimeoutTest.java
+++ b/jetty-server/src/test/java/org/eclipse/jetty/server/ServerConnectorTimeoutTest.java
@@ -37,6 +37,7 @@ import org.eclipse.jetty.server.session.SessionHandler;
 import org.eclipse.jetty.toolchain.test.AdvancedRunner;
 import org.eclipse.jetty.util.IO;
 import org.eclipse.jetty.util.log.StacklessLogging;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
 import org.hamcrest.Matchers;
 import org.junit.Assert;
 import org.junit.Before;
@@ -134,7 +135,7 @@ public class ServerConnectorTimeoutTest extends ConnectorTimeoutTest
     @Test
     public void testHttpWriteIdleTimeout() throws Exception
     {
-        _httpConfiguration.setBlockingTimeout(500);
+        _httpConfiguration.setIdleTimeout(500);
         configureServer(new AbstractHandler.ErrorDispatchHandler()
         {
             @Override

